### PR TITLE
Update survey proto to support data sharing agreement

### PR DESCRIPTION
**Pull request by nwkotto**
Part 1 for https://github.com/google/ground-platform/issues/1856

### CODE REVIEW

File: proto/src/google/ground/v1beta1/survey.proto [hunk 1]
```diff
@@ -37,6 +37,18 @@ message Survey {
 
   // Required. User id of the survey owner.
   string owner_id = 5;
+
+  // Properties related to sharing of survey data.
+  message DataSharing {
```
Comment (sufyanAbbasi) [inline proto/src/google/ground/v1beta1/survey.proto:42]: Maybe `DataSharingConsent dataSharingConsent` and `DataSharingConsentType`? Nouns for messages are preferable and making the "consent" part explicit differentiates it from ACLs.

**Comment by sufyanAbbasi** [comment]
Awesome! And I believe we will also want an associated response field on the `Submission` too, since we'll want to save the user's response to the DB in case the survey organizers change the sharing consent after the fact. Maybe just an enum will suffice?

**Comment by gino-m** [comment]
> Awesome! And I believe we will also want an associated response field on the `Submission` too, since we'll want to save the user's response to the DB in case the survey organizers change the sharing consent after the fact. Maybe just an enum will suffice?

Not sure about storing Submission - a Firestore query would scan all submissions to find out if one has the consent bit. That said, Survey isn't editable.. maybe a new collection specifically for this purpose? For a first plass we'll just ask once per session per survey, so nothing needs to be stored.

File: proto/src/google/ground/v1beta1/survey.proto (hunk 1 above)
Comment (nwkotto) [reply proto/src/google/ground/v1beta1/survey.proto]: Sounds good! Updated the proto names per that suggestion.

**Comment by nwkotto** [comment]
> > Awesome! And I believe we will also want an associated response field on the `Submission` too, since we'll want to save the user's response to the DB in case the survey organizers change the sharing consent after the fact. Maybe just an enum will suffice?
> 
> Not sure about storing Submission - a Firestore query would scan all submissions to find out if one has the consent bit. That said, Survey isn't editable.. maybe a new collection specifically for this purpose? For a first plass we'll just ask once per session per survey, so nothing needs to be stored.

So are we saying we're ok to not include any submission bit about consent being agreed to? Or how are we expecting that to be captured?

File: proto/src/google/ground/v1beta1/survey.proto [hunk 2]
```diff
@@ -53,3 +65,15 @@ enum Role {
   // User may also manage the survey, update locations of interest, etc.
   SURVEY_ORGANIZER = 3;
 }
+
+// Defines the type of data sharing consent agreement the user can see.
+enum DataSharingConsentType {
```
Comment (gino-m) [inline proto/src/google/ground/v1beta1/survey.proto:70]: For consistency, we can move this inside `DataShareConsent` above the field where it's used and rename it to `Type`.

File: proto/src/google/ground/v1beta1/survey.proto [hunk 3]
```diff
@@ -37,6 +37,18 @@ message Survey {
 
   // Required. User id of the survey owner.
   string owner_id = 5;
+
+  // Properties related to data sharing consent.
+  message DataSharingConsent {
```
Comment (gino-m) [inline proto/src/google/ground/v1beta1/survey.proto:42]: I think "data sharing terms" would be more correct than "consent" here. Consent can be reserved for once the data collectors actually agree to the terms.

File: proto/src/google/ground/v1beta1/survey.proto [hunk 4]
```diff
@@ -37,6 +37,18 @@ message Survey {
...
+    // Type of data sharing consent agreement the user will see.
+    DataSharingConsentType type = 1;
+
+    // Text of custom data sharing consent terms.
+    string custom_text = 2;
+  }
+
+  // Defines how data sharing consent agreemnent is displayed to users.
+  DataSharing dataSharingConsent = 6;
```
Comment (gino-m) [inline proto/src/google/ground/v1beta1/survey.proto:51]: s/DataSharing/DataSharingTerms/

Also, dataSharingConsent -> lower_snake_case.

**Comment by gino-m** [comment]
> > > Awesome! And I believe we will also want an associated response field on the `Submission` too, since we'll want to save the user's response to the DB in case the survey organizers change the sharing consent after the fact. Maybe just an enum will suffice?
> > 
> > 
> > Not sure about storing Submission - a Firestore query would scan all submissions to find out if one has the consent bit. That said, Survey isn't editable.. maybe a new collection specifically for this purpose? For a first plass we'll just ask once per session per survey, so nothing needs to be stored.
> 
> So are we saying we're ok to not include any submission bit about consent being agreed to? Or how are we expecting that to be captured?

Sorry, I think I misunderstood the original question. Adding a bit to submissions as to which consent was given is a good idea. For now the Android client will need to store that consent was given on the first data collection flow locally, in the future we may store that in the remote db as well.

**Review by gino-m — APPROVED** [review]
I made a few small adjustments, LGTM!

**Pull request merged**